### PR TITLE
broom: init at 0.3.0

### PR DESCRIPTION
**Pull request by a-camarillo**
## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: maintainers/maintainer-list.nix [hunk 1]
```diff
@@ -423,6 +423,12 @@
     github = "abysssol";
     githubId = 76763323;
   };
+  a-camarillo = {
```
Comment (vinnymeller) [inline maintainers/maintainer-list.nix:426]: from the github checks that ran:

 trace: maintainer acairncross (line 432) should be listed after abysssol, not after a-camarillo (line 426)

**Review by vinnymeller — CHANGES_REQUESTED** [review]
The derivation and commit messages look good. Just the maintainer list order failing check + below comments:

## nixpkgs-review

Just because PRs often take a while to get to, I recommend running [nixpkgs-review](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#tested-compilation-of-all-pkgs-that-depend-on-this-change-using-nixpkgs-review) yourself - other people can always verify themselves later as they please.

In this case, running`nix run nixpkgs#nixpkgs-review -- pr 306057` gives me...

Result of `nixpkgs-review pr 306057` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)

...this indicates that this PR isn't actually adding the package as expected

## Directory structure

With this derivation defined where it is currently, you'd need an entry in [this file](https://github.com/NixOS/nixpkgs/blob/master/pkgs/top-level/all-packages.nix) to expose the package.

But _most_ new packages should just go in pkgs/by-name/, so this derivation would go in `pkgs/by-name/br/broom/package.nix`

With the file there, you don't need to add an entry to a top-level file, it'll get picked up automagically

This new(er) structure comes from [RFC 140](https://github.com/NixOS/rfcs/blob/master/rfcs/0140-simple-package-paths.md)

**Comment by a-camarillo** [comment]
Result of `nix run nixpkgs#nixpkgs-review -- pr 306057` on x86_64-linux

1 package built:
broom

**Comment by vinnymeller** [comment]
@a-camarillo instead of the 3rd commit for the fix, put the fixed versions in their respective commits so you still just have the (fixed) 1st 2 commits

**Comment by a-camarillo** [comment]
@vinnymeller when you have a chance to re-review, I fixed the commits and all tests are passing

**Comment by vinnymeller** [comment]
@a-camarillo i think last thing would be to just swap the order of your 2 commits so adding yourself as a maintainer comes 1st. while it doesn't really matter, it's technically less correct & easy enough to do

in case you don't know how to do it easily, you can just `git rebase -i HEAD~2`

and then swap the order of the commits:

```
pick 4e9d5be07beb broom: init at 0.3.0
pick 43cbf30ec808 maintainers: add a-camarillo
```
to
```
pick 43cbf30ec808 maintainers: add a-camarillo
pick 4e9d5be07beb broom: init at 0.3.0
```

save force push & done

**Comment by a-camarillo** [comment]
@vinnymeller alright changes are made. Appreciate the patience and help by the way, it really means a lot.

**Review by vinnymeller — APPROVED** [review]

**Comment by vinnymeller** [comment]
> Appreciate the patience

of course :smiley: though id say a _majority_ of the time I say something LGTM somebody who knows better than me comes and finds something they want changed also - so maybe expect that

and that step is how I learn :laughing:

**Review by nicoonoclaste — APPROVED** [review]

**Comment by a-camarillo** [comment]
@nbraud I don't mind adding the updateScript, would I just add `passthru.updateScript = nix-update-script { };`. 

Also not entirely sure about the approach for making this change. Would I just create a new branch with a commit for adding the updateScript?

**Pull request merged**